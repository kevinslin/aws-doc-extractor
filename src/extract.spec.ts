import { extractFromAWSDocs } from './index'; // Update with the correct path to your function
import fs from 'fs';

describe('extractFromAWSDocs', () => {
    it('should extract entities according to the given conditions', () => {
        const testMarkdown = `# Section 1

Some content.

**Note**

This is a note.

This is a paragraph following the note.

# Section 2

**Note**

This is a paragraph following the second note:

This is a second paragraph after the colon.

This is not captured.
`;

        fs.writeFileSync('test.md', testMarkdown);

        const expectedEntities = [
            {
                position: {
                    start: { line: 7, column: 1, offset: 6 },
                    end: { line: 7, column: 1, offset: 6 },
                    indent: 0,
                },
                content: 'This is a note.',
                header: 'Section 1',
            },
            {
                position: {
                    start: { line: 15, column: 1, offset: 14 },
                    end: { line: 17, column: 1, offset: 16 },
                    indent: 0,
                },
                content:
                    'This is a paragraph following the second note:\n\nThis is a second paragraph after the colon.',
                header: 'Section 2',
            },
        ];

        const entities = extractFromAWSDocs('test.md');
        expect(entities).toMatchSnapshot()
        expect(entities).toEqual(expectedEntities);

        fs.unlinkSync('test.md');
    });
});
