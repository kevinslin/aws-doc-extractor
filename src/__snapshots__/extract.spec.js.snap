// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractFromAWSDocs should extract entities according to the given conditions 1`] = `
[
  {
    "content": "This is a paragraph following the second note:",
    "header": "Section 2",
    "position": {
      "end": {
        "column": 1,
        "line": 14,
        "offset": 13,
      },
      "indent": 0,
      "start": {
        "column": 1,
        "line": 14,
        "offset": 13,
      },
    },
  },
]
`;
